al_mlp/ensemble_calc.py

```diff
@@ -98,23 +107,28 @@ def train_and_combine(args_tuple):
             check_path = trainer.cp_dir
             trainer = AtomsTrainer()
             trainer.load_pretrained(checkpoint_path=check_path)
-            trained_calc = DeltaCalc((trainer.get_calc(),base_calc),"add",refs)
+            trained_calc = DeltaCalc((trainer.get_calc(), base_calc), "add", refs)
             return trained_calc
 
-        #split ensemble sets into separate tuples, clone: trainer, base calc and add to tuples, add: refs to tuples
+        # split ensemble sets into separate tuples, clone: trainer, base calc and add to tuples, add: refs to tuples
         tuples = []
         random.seed(trainer.config["cmd"]["seed"])
-        randomlist = [random.randint(0,4294967295) for set in ensemble_sets]
+        randomlist = [random.randint(0, 4294967295) for set in ensemble_sets]
         for i in range(len(ensemble_sets)):
             set = ensemble_sets[i]
-            trainer_copy = AtomsTrainer(copy.deepcopy(trainer.config))
-            trainer_copy.config["cmd"]["seed"] = randomlist[i]
-            trainer_copy.load_rng_seed()
+
+            copy_config = copy.deepcopy(trainer.config)
+            copy_config["cmd"]["seed"] = randomlist[i]
+            copy_config["cmd"]["identifier"] = copy_config["cmd"]["identifier"] + str(
+                uuid.uuid4()
```

Comment:
What is the purpose of the identifier? Does this have to do with ensuring each member of the ensemble references its own unique directory, or is this just so that they may be differentiated?

Reply:
Yep, the subdirectory in "checkpoints" is then differentiated between individual trainers